mne/minimum_norm/tests/test_inverse.py

```diff
@@ -709,6 +710,63 @@ def test_io_inverse_operator():
     _compare(inv_prep, inv_prep_prep)
 
 
+@testing.requires_testing_data
+def test_apply_inverse_cov():
+    """Test MNE with precomputed inverse operator on cov."""
+    raw = read_raw_fif(fname_raw, preload=True)
+    # # use 10 sec of data
+    # raw.crop(0, 10)
+    raw.filter(1, None)
+    label_lh = read_label(fname_label % 'Aud-lh')
+    inverse_operator = read_inverse_operator(fname_full)
+
+    data_cov = compute_raw_covariance(raw, tmin=raw.times[0],
+                                      tmax=raw.times[-1], tstep=None)
+
+    with pytest.raises(ValueError, match='has not been prepared'):
+        apply_inverse_cov(data_cov, raw.info, 1, inverse_operator,
+                          lambda2=lambda2, prepared=True)
+
+    first_test_flag = True
+    # pick_ori='normal' is not tested because apply_inverse_raw returns
+    # unsigned magnitudes which can't be compared with apply_inverse_cov output
+    # cov_scale compensates for the 3 orientations returned with
+    # pick_ori='vector'
+    for method in INVERSE_METHODS:
+        this_inv_op = prepare_inverse_operator(inverse_operator, nave=1,
+                                               lambda2=lambda2, method=method)
+        for pick_ori, cov_scale in zip(['normal', 'vector'], [1, 3]):
+            stc_raw = apply_inverse_raw(raw, this_inv_op, lambda2, method,
+                                        label=label_lh, nave=1,
+                                        pick_ori=pick_ori, prepared=True)
+
+            stc_cov = apply_inverse_cov(data_cov, raw.info, 1, this_inv_op,
+                                        method=method, pick_ori=pick_ori,
+                                        label=label_lh, prepared=True,
+                                        lambda2=lambda2, dB=False)
+
+            # only do this test once
+            if first_test_flag:
+                first_test_flag = False
+                stc_cov_db = apply_inverse_cov(data_cov, raw.info, 1,
+                                               this_inv_op, method=method,
+                                               pick_ori=pick_ori, dB=True,
+                                               label=label_lh, prepared=True,
+                                               lambda2=lambda2)
+                assert (stc_cov.subject == 'sample')
+                assert (stc_cov_db.subject == 'sample')
+                assert_array_almost_equal(stc_cov_db.data,
+                                          10 * np.log10(stc_cov.data))
+
+            n_sources = stc_raw.data.shape[0]
+            raw_data = stc_raw.data.reshape(n_sources, -1)
+            exp_res = cov_scale * np.diag(np.cov(raw_data))
+            assert_allclose(exp_res, stc_cov.data.ravel(), atol=1e-26,
+                            rtol=1e-3,
```

Comment:
Why do we need such a high `rtol`? Since the inverse operator is a linear operator, shouldn't the result of applying it to `raw` and then computing power be exactly the same as applying it to the covariance of `raw`?

Reply:
I'm not sure about this. there seems to be a sight scaling difference. for instance using `cov_scale = 1.0001664968775532` for `pick_ori='normal'` will pass with an 'rtol=1e-8' but the best I can get for `pick_ori='vector' ` is an `rtol=1e-5`

Similarly I'm not sure where this scaling factor comes from. I suspect it is subtle differences in the covariance computations np.cov vs sckit code used by mne.covariance. if you have an ideas or suggestions feel free to let me know.

Reply:
FWIW `np.cov` has `bias=False` by default, meaning their normalization is by `N-1` instead of `N`. MNE probably uses `N`.

Reply:
(in other words, see if doing `np.cov(..., ddof=0)` helps at all, as `ddof` seems to be preferred way of changing this behavior in `np.cov`)

Reply:
I was thinking along a similar line but that did not solve the issue.

is there a trivial way to make an `mne.Covariance` object from a data matrix?

Reply:
what about explicitly computing power as: `np.sum(raw_data ** 2, axis=1) / raw_data.shape[1]`?